python/ray/data/dataset.py

```diff
@@ -3414,6 +3414,15 @@ def __repr__(self) -> str:
     def __str__(self) -> str:
         return repr(self)
 
+    def __bool__(self) -> bool:
+        return True
```

Comment:
Could you add a comment here? I don't think readers would understand why we implement `__bool__` without reading this thread.

Reply:
Yup, seems like a good idea too.